schema_test.go

```diff
@@ -181,3 +181,20 @@ func TestLoadersWithInvalidPattern(t *testing.T) {
 		assert.NotNil(t, err, "expected error loading invalid pattern: %T", l)
 	}
 }
+
```

Comment:
If I change the test to the following it doesn't work.

```go
const schemaWithRef = `{"$ref": "http://url-to-external-schema"}`
const localSchema = `{"type": "string"}`

func TestLocalSchema(t *testing.T) {
	content := `"this is a string"`
	// s := NewReferenceLoader("http://local/schemaWithRef")
	s := NewStringLoader(schemaWithRef)
	d := NewBytesLoader([]byte(content))

	// s.AddSchema("http://local/schemaWithRef", NewBytesLoader([]byte(schemaWithRef)))

	// Use cached local copy of external schema
	s.AddSchema("http://url-to-external-schema", NewBytesLoader([]byte(localSchema)))

	res, err := Validate(s, d)
	assert.Nil(t, err)
	assert.True(t, res.Valid())
}
```

This currently is because all other loaders use `DefaultJSONLoaderFactory` instead of the `FileSystemJSONLoaderFactory` which has the schema caching.  I sort of get why there is a distinction, but it functionally doesn't really help. I'll probably dive in myself in the coming days to see if I can come up with a solution.

Reply:
👍 

I took a swing at solving, but figured there must be a better way if you know the codebase.